### ISSUE TITLE
Nexus terraform proxy

### ISSUE THREAD

**Issue by realRunlo** [sonatype/nexus-public#481]
Hi,

I have configured a raw repository in Nexus as a Terraform provider mirror, and it works well. However, this solution requires me to implement an automation to manage the process of uploading providers to the repository, which I would like to avoid.

To address this, I set up a raw proxy in Nexus to serve as a proxy for https://registry.terraform.io/. The problem arises when I run the following command:

`HTTPS_PROXY=https://nexus-instance/repositories/tf-proxy terraform init`

I receive the error: malformed HTTP response. This is expected since the proxy is not configured for this specific use case.

For example, when `terraform init` is executed, it starts by requesting https://registry.terraform.io/.well-known/terraform.json. If I run:

`curl https://nexus-instance/.well-known/terraform.json`

It works correctly, and the response gets cached.

The ideal solution would be to have a dedicated "Terraform proxy" in Nexus or to have the capability to perform additional configurations on the raw proxy.

Nexus Version: OSS 3.72.0 on ECS.

Best regards,
Great job, by the way!

**Comment by gracecllee** [comment]
Hi @realRunlo , thanks for submitting this, and really appreciate the pat on the back!  We love that our community is actively using our product to its full potential. As for Terraform format support, we'd love to learn more about your usage and what would be helpful. Would you be interested in talking to our product team to provide feedback? Thanks!

**Comment by realRunlo** [comment]
I'm open to talk to your product team, thanks.

**Comment by dmpe** [comment]
Well, isn't that the same as in #415 #120 ? JFrog has it https://jfrog.com/help/r/jfrog-artifactory-documentation/configure-terraform-provider-registry-with-artifactory-using-terraform-client

**Comment by realRunlo** [comment]
Yes jfrog has it but i was looking for the same feature in nexus, thanks anyway.

**Comment by dmpe** [comment]
No idea, my reply was to @gracecllee :) ... Nexus should finally make it work. There is demand and there is competitor disadvantage of not having.

**Comment by BenjaminSchweizer** [comment]
terraform provider support would be awesome

**Comment by mbounaceur** [comment]
I confirm, terraform support would be awesome. It can very useful for many people that only allow internet traffic through Nexus

**Comment by hoazgazh** [comment]
`[2024-12-30T03:22:36.566] [ERROR] default - ╷
│ Error: Failed to query available provider packages
│ 
│ Could not retrieve the list of available versions for provider
│ gitlabhq/gitlab: could not connect to registry.terraform.io: failed to
│ request discovery document: Get
│ "https://registry.terraform.io/.well-known/terraform.json": read tcp
│ xxx.xxxx.xxx:443: read: connection reset by peer
╵
gitlab-provider  ╷
                 │ Error: Failed to query available provider packages
                 │ 
                 │ Could not retrieve the list of available versions for provider
                 │ gitlabhq/gitlab: could not connect to registry.terraform.io: failed to
                 │ request discovery document: Get
                 │ "https://registry.terraform.io/.well-known/terraform.json": read tcp
                 │ xxx.xxx.xx.xx"443: read: connection reset by peer
                 ╵
                 
⠏  Processing
External Error: Stack failed to plan: gitlab-provider. Please check the logs for more information.`

Same here, I need to run terraform init from an internal network or private network. I also have a Nexus repository proxy to the Terraform registry. Hopefully, this feature can be completed soon.

**Comment by brontolinux** [comment]
We'd definitely need a terraform registry proxy/cachein Nexus. Running Nexus Pro 3.74, and it's appalling that this feature is not even on the radar yet...

**Comment by mrprescott** [comment]
It's definitely on our radar, and we have plans to implement it in a coming version. I don't have specific timing to share at the moment, but we are aware that there is healthy demand for this.

**Comment by brontolinux** [comment]
> It's definitely on our radar, and we have plans to implement it in a coming version. I don't have specific timing to share at the moment, but we are aware that there is healthy demand for this.

Hi @mrprescott,

Thanks for the update - I appreciate that Sonatype is monitoring this issue and taking user feedback into account.

That said, "on our radar," "we have plans," and "no specific timing" usually mean this could take months, or beyond 2025. While we’re staying with your platform for now, the lack of this feature is becoming more of a concern for our team.

I hope it gets the priority it deserves. Looking forward to updates when there’s more to share!

Cheers,
-- M

**Comment by raZorTT** [comment]
Any updates or announcements that can be made about TF proxy support?

**Comment by rosscdh-tpg** [comment]
Any updates on this request please?

**Comment by a2xzy** [comment]
Hi @gracecllee,

Do you have any update on this please? This would be helpful to us too.

**Comment by rudesome** [comment]
@mrprescott ; any update on this feature request? would like to use Nexus as a Terraform module / provider backend

**Comment by tuananh** [comment]
IIRC I hear it's coming soon in Q1 2026.

**Comment by brontolinux** [comment]
> IIRC I hear it's coming soon in Q1 2026.

Uhm... do you recall where you did hear it? Because I have done some digging with DuckDuckGo, and found nothing...

Also, is this something you could confirm @mrprescott ?

**Comment by miguelangat** [comment]
Yes, Terraform proxy support is coming in Q1 with our January release!

**Comment by laxmi-st** [comment]
Terraform Proxy support is now available in [Nexus Repository 3.88.0](https://help.sonatype.com/en/sonatype-nexus-repository-3-88-0-release-notes.html#support-for-proxy-terraform-repositories)
